src/client.rs

```diff
@@ -47,21 +49,123 @@ pub struct Client {
     inner: Arc<ClientRef>,
 }
 
-impl Client {
-    /// Constructs a new `Client`.
-    pub fn new() -> ::Result<Client> {
-        let mut client = try_!(new_hyper_client());
-        client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
-        Ok(Client {
+impl From<NativeTlsClient> for Client {
```

Comment:
If we did the conservative change like I suggest in the review comment, we'd probablt want to remove this impl.